### PR TITLE
add cilium clusterwide network policies support

### PR DESCRIPTION
**Pull request by maxi-cit**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/17620

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
container: added `enable_cilium_clusterwide_network_policy` field to `google_container_cluster` resource
```

### CODE REVIEW

File: mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb [hunk 1]
```diff
@@ -2252,13 +2257,14 @@ func resourceContainerClusterCreate(d *schema.ResourceData, meta interface{}) er
...
-			DefaultSnatStatus:         expandDefaultSnatStatus(d.Get("default_snat_status")),
-		    DatapathProvider:          d.Get("datapath_provider").(string),
-			PrivateIpv6GoogleAccess:   d.Get("private_ipv6_google_access").(string),
-			EnableL4ilbSubsetting:     d.Get("enable_l4_ilb_subsetting").(bool),
-			DnsConfig:                 expandDnsConfig(d.Get("dns_config")),
-			GatewayApiConfig:          expandGatewayApiConfig(d.Get("gateway_api_config")),
+			EnableIntraNodeVisibility:            d.Get("enable_intranode_visibility").(bool),
+			DefaultSnatStatus:                    expandDefaultSnatStatus(d.Get("default_snat_status")),
+		    DatapathProvider:                     d.Get("datapath_provider").(string),
+			EnableCiliumClusterwideNetworkPolicy: d.Get("enable_cilium_clusterwide_network_policy").(bool),
```
Comment (maxi-cit) [inline mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb:2263]: This field needs to be supported in the golang lib in order to make it work

**Comment by maxi-cit** [comment]
[This library](https://pkg.go.dev/google.golang.org/api@v0.170.0/container/v1beta1#NetworkConfig) needs to get enhanced to include the `EnableCiliumClusterwideNetworkPolicy` and `DesiredEnableCiliumClusterwideNetworkPolicy` attribues

**Comment by maxi-cit** [comment]
FYI I bumped the google.golang.org/api lib using 

`go get -u google.golang.org/api@latest`

because it contains the support for the required field for this PR

**Comment by maxi-cit** [comment]
Hello @NickElliot, I think I am ready for another round of review.

I added two tests that showcase the support of the field for standar cluster and autopilot mode.

PD: should I add the golang lib bumping within the changelog?

**Comment by NickElliot** [comment]
Can you resolve the newly arisen conflicts? Sorry I keep trying to review this today but it keeps needing the test reran and now this came up 😅

**Comment by maxi-cit** [comment]
Hello @NickElliot , it makes sense since all contribute to the same go dependency. So I tried to keep the latest versions of all libraries. go.sum should be updated as well

**Comment by maxi-cit** [comment]
Hello @NickElliot, it seems that tests are ok. Is there anything that is blocking this merge?

**Pull request merged**